### DISCUSSION TITLE
WeatherService bei Erweiterung weewx-DWD

### DISCUSSION THREAD

**Discussion by Ainadilion** [roe-dl/weewx-DWD#73] [Q&A]
Ich habe noch eine frühere Erweiterung ohne WeatherService, welche Schritte sind erforderlich, um WeatherService zu nutzen? 

Weewx 4.10.2

**Comment by roe-dl** [comment]
Vor allen anderen Maßnahmen bitte immer ein Backup anfertigen!

Man kann die aktuelle Version einfach nach Anleitung drüberinstallieren. Dinge, die vorher funktioniert haben, sollten auch weiter funktionieren. Die neuen Funktionen können dann konfiguriert werden. Nach Konfigurationsänderungen muß WeeWX neu gestartet werden.

* [Anleitung](https://github.com/roe-dl/weewx-DWD/blob/master/README-de.md)
* [Wiki](https://github.com/roe-dl/weewx-DWD/wiki)

Bei konkreten Problemen bitte fragen.

**Comment by Ainadilion** [comment]
Hallo,

beim Versuch die Erweiterung zu installieren, erhalte ich folgende Meldungen:

***@***.***:/etc/weewx $ sudo wee_extension --install weewx-dwd.zip
Request to install 'weewx-dwd.zip'
Extracting from zip archive weewx-dwd.zip
{'WEEWX_ROOT': '/', 'CONFIG_ROOT': '/etc/weewx', 'BIN_ROOT': '/usr/share/weewx', 'USER_ROOT': '/usr/share/weewx/user', 'EXT_ROOT': '/usr/share/weewx/user/installer', 'SKIN_ROOT': '/etc/weewx/skins'}
Traceback (most recent call last):
  File "/usr/share/weewx/wee_extension", line 92, in <module>
    main()
  File "/usr/share/weewx/wee_extension", line 84, in main
    ext.install_extension(options.install)
  File "/usr/share/weewx/weecfg/extension.py", line 130, in install_extension
    self.install_from_dir(extension_dir)
  File "/usr/share/weewx/weecfg/extension.py", line 216, in install_from_dir
    save_config |= installer.configure(self)
  File "/var/tmp/weewx-DWD-master/install.py", line 84, in configure
KeyError: 'USER_DIR'

Was wäre hier zu tun?

Gruß
Steffen

Gesendet von Outlook für iOS<https://aka.ms/o0ukef>
________________________________
Von: roe-dl ***@***.***>
Gesendet: Sunday, August 25, 2024 9:20:10 PM
An: roe-dl/weewx-DWD ***@***.***>
Cc: Ainadilion ***@***.***>; Author ***@***.***>
Betreff: Re: [roe-dl/weewx-DWD] WeatherService bei Erweiterung weewx-DWD (Discussion #73)


Vor allen anderen Maßnahmen bitte immer ein Backup anfertigen!

Man kann die aktuelle Version einfach nach Anleitung drüberinstallieren. Dinge, die vorher funktioniert haben, sollten auch weiter funktionieren. Die neuen Funktionen können dann konfiguriert werden. Nach Konfigurationsänderungen muß WeeWX neu gestartet werden.

  *   Anleitung<https://github.com/roe-dl/weewx-DWD/blob/master/README-de.md>
  *   Wiki<https://github.com/roe-dl/weewx-DWD/wiki>

Bei konkreten Problemen bitte fragen.

**Reply by roe-dl** [reply]
Oh, oh, da haben die Änderungen für WeeWX 5.0 die Installation für WeeWX 4.10 kaputt gemacht. Ich werde es korrigieren, aber derweilen bitte wie folgt vorgehen:
* `weewx-dwd.zip` auspacken. Das landet typischerweise in einem Unterverzeichnis `weewx-dwd`
* in `install.py` in Zeile 84 das Textstück `,engine.root_dict['USER_DIR']` ersatzlos entfernen. Einschließlich des Kommas. Die Zeile lautet dann nur noch:
  ```Python
  user_root = engine.root_dict.get('USER_ROOT')
  ```
* die Installation nun mit `sudo wee_extension --install weewx-dwd` (ohne `.zip` am Ende) durchführen.

**Comment by Ainadilion** [comment]
Hallo,
nach der Anleitung vorgegangen, nun erscheinen folgende Meldungen:

***@***.***:/etc/weewx/weewx-DWD-master $ sudo wee_extension --install weewx-dwd
Request to install 'weewx-dwd'
Traceback (most recent call last):
  File "/usr/share/weewx/wee_extension", line 92, in <module>
    main()
  File "/usr/share/weewx/wee_extension", line 84, in main
    ext.install_extension(options.install)
  File "/usr/share/weewx/weecfg/extension.py", line 139, in install_extension
    raise InstallError("Extension '%s' not found." % extension_path)
weecfg.extension.InstallError: Extension 'weewx-dwd' not found.

Gruß
Steffen

Gesendet von Outlook für iOS<https://aka.ms/o0ukef>
________________________________
Von: roe-dl ***@***.***>
Gesendet: Monday, August 26, 2024 2:34:02 PM
An: roe-dl/weewx-DWD ***@***.***>
Cc: Ainadilion ***@***.***>; Author ***@***.***>
Betreff: Re: [roe-dl/weewx-DWD] WeatherService bei Erweiterung weewx-DWD (Discussion #73)


Oh, oh, da haben die Änderungen für WeeWX 5.0 die Installation für WeeWX 4.10 kaputt gemacht. Ich werde es korrigieren, aber derweilen bitte wie folgt vorgehen:

  *   weewx-dwd.zip auspacken. Das landet typischerweise in einem Unterverzeichnis weewx-dwd
  *   in install.py in Zeile 84 das Textstück ,engine.root_dict['USER_DIR'] ersatzlos entfernen. Einschließlich des Kommas. Die Zeile lautet dann nur noch:

user_root = engine.root_dict.get('USER_ROOT')

  *   die Installation nun mit sudo wee_extension --install weewx-dwd (ohne .zip am Ende) durchführen.

**Reply by roe-dl** [reply]
```shell
cd ..
sudo wee_extension --install weewx-DWD-master
```

Offenbar hieß das Verzeichnis beim Entpacken nicht `weewx-dwd` sondern `weewx-DWD-master`, wenn ich das richtig interpretiere. Aktuelles Verzeichnis sollte jedenfalls das Verezeichnis sein, wo die ursprünglich die ZIP-Datei hineingeladen wurde. Das scheint mir hier `/etc/weewx` gewesen zu sein. Habe ich das richtig gesehen?

**Comment by Ainadilion** [comment]
cd .. war die Lösung, vielen lieben Dank

Gesendet von Outlook für iOS<https://aka.ms/o0ukef>
________________________________
Von: roe-dl ***@***.***>
Gesendet: Monday, August 26, 2024 3:01:16 PM
An: roe-dl/weewx-DWD ***@***.***>
Cc: Ainadilion ***@***.***>; Author ***@***.***>
Betreff: Re: [roe-dl/weewx-DWD] WeatherService bei Erweiterung weewx-DWD (Discussion #73)


cd ..
sudo wee_extension --install weewx-DWD-master

Offenbar hieß das Verzeichnis beim Entpacken nicht weewx-dwd sondern weewx-DWD-master, wenn ich das richtig interpretiere. Aktuelles Verzeichnis sollte jedenfalls das Verezeichnis sein, wo die ursprünglich die ZIP-Datei hineingeladen wurde. Das scheint mir hier /etc/weewx gewesen zu sein. Habe ich das richtig gesehen?

**Reply by roe-dl** [reply]
Jetzt bin ich erst einmal erleichtert.

**Comment by Ainadilion** [comment]
Ich auch, danke für die schnelle Unterstützung

Gesendet von Outlook für iOS<https://aka.ms/o0ukef>
________________________________
Von: roe-dl ***@***.***>
Gesendet: Monday, August 26, 2024 3:21:32 PM
An: roe-dl/weewx-DWD ***@***.***>
Cc: Ainadilion ***@***.***>; Author ***@***.***>
Betreff: Re: [roe-dl/weewx-DWD] WeatherService bei Erweiterung weewx-DWD (Discussion #73)


Jetzt bin ich erst einmal erleichtert.
